src/hostapi/pulseaudio/pa_linux_pulseaudio.c

```diff
@@ -1311,3 +1334,82 @@ double GetStreamCpuLoad( PaStream * s )
 
     return PaUtil_GetCpuLoad( &stream->cpuLoadMeasurer );
 }
+
+/** Extensions */
+static void RenameStreamCb(pa_stream *s, int success, void *userdata)
+{
+    /* Currently does nothing but signal the caller. */
+    PaPulseAudio_Stream *l_ptrStream = (PaPulseAudio_Stream *) userdata;
+    pa_threaded_mainloop_signal( l_ptrStream->mainloop,
+                                 0 );
+}
+
+PaError PaPulseAudio_RenameSource( PaStream *s, const char *streamName )
+{
+    PaPulseAudio_Stream *stream = (PaPulseAudio_Stream *) s;
+    PaError result = paNoError;
+    pa_operation *op = NULL;
+
+    /* Reallocate stream name in memory. */
+    char *newStreamName = (char*)PaUtil_AllocateMemory(strlen(streamName));
```

Comment:
I think we should use strnlen like (paranoid I know) with PAPULSEAUDIO_MAX_DEVICENAME just in case.

Reply:
What max size would be reasonable in your opinion? 128 bytes maybe?

Reply:
I think 'PAPULSEAUDIO_MAX_DEVICENAME' define should be used which now is 1024 which is overkill but I think I'll drop it to 256 bytes. Names are machine created so they can be very long.

Reply:
+1 for '\0'

Reply:
Updated to use sizeof() so +1 for '\0' should already be included (and to use strnlen/strncpy with PAPULSEAUDIO_MAX_DEVICENAME).